### DISCUSSION TITLE
[Feature Request] Trace source of atom updates

### DISCUSSION THREAD

**Discussion by kevinschaich** [jotaijs/jotai-devtools#135] [Ideas]
When your state becomes large enough, effects tend to overlap – especially in applications with a lot of visual/input interactions (mouse movement, tracking positions of elements on the screen, dragging and dropping, etc). This can lead to render loops that can be hard to track down. If you have 10 buttons/sliders/keyboard shortcuts that all perform the same state update, it's really time consuming to figure out which one of them is causing issues.

[Redux devtools has a nice feature](https://github.com/reduxjs/redux-devtools/blob/main/extension/docs/Features/Trace.md) that lets you "trace" the source of updates:

![Trace feature](https://github.com/jotaijs/jotai-devtools/assets/9244728/40d240ea-7ebf-44f9-9a61-2c855d56f87c)

I think this would be a nice addition to the devtools, perhaps in the Time Travel tab behind a (default off) feature flag or setting.

**Comment by arjunvegda** [comment]
Thanks for reaching out! 

This feature has [been on my mind](https://github.com/pmndrs/jotai/issues/931) for a while now. Perhaps something I can look into after adding support for Jotai store v2. Open to PRs if you have ideas as well 🙌

**Comment by kevinschaich** [comment]
Hey @arjunvegda wanted to check in. Do you have any ideas on how we'd implement this? 

Happy to do what I can to contribute

**Comment by kevinschaich** [comment]
#168
